lib/core/test/unit/Cardano/Wallet/Primitive/Types/TokenMapSpec.hs

```diff
@@ -255,6 +261,19 @@ prop_subtract_invariant m1 m2 = property $
   where
     Just result = TokenMap.subtract m1 m2
 
+prop_difference_zero :: TokenMap -> Property
+prop_difference_zero x =
+    x `difference` TokenMap.empty === x
+
+prop_difference_leq :: TokenMap -> TokenMap -> Property
+prop_difference_leq x y = property $
+    x `difference` y `leq` x
+
+prop_difference_add :: TokenMap -> TokenMap -> Property
+prop_difference_add x y = property $
+    (x `difference` y) `TokenMap.add` y
+        === x
```

Comment:
I think in general this isn't true, unless this is also true:

```hs
y `leq` x
```

Reply:
Yup, I just noticed.

Reply:
This seems to be true though:
```haskell
prop_difference_add :: TokenBundle -> TokenBundle -> Property
prop_difference_add x y = property $
    ((x `difference` y) `add` y)
        `geq` x
  where
    geq a b = not $ leq a b
```

Reply:
The definition of `geq` should also handle the equality case:

```hs
geq a b = a == b || (not $ leq a b)
```

Reply:
I went with
```haskell
b `leq` a
```